source/backend/cpu/compute/BlstmComputer.cpp

```diff
@@ -0,0 +1,346 @@
+
+  // copy weight data
+  for (int b = 0; b < (mBidirectional ? 2 : 1); b++) {
+    // b = 0 -> forward, b = 1 -> backward
+    for (int i = 0 + b * 12; i < 4 + b * 12; i++) {
+      MNN_ASSERT(weightsVec[i]->dimensions() == 2);
+      MNN_ASSERT(weightsVec[i]->buffer().dim[0].extent == mInDim);
+      MNN_ASSERT(weightsVec[i]->buffer().dim[1].extent == mStateSize);
+      trimTensor(weightsVec[i].get(), mWeights[i].get());
+    }
+    for (int i = 4 + b * 12; i < 8 + b * 12; i++) {
+      // Ui, Un, Uf, Uo
+      MNN_ASSERT(weightsVec[i]->dimensions() == 2);
+      MNN_ASSERT(weightsVec[i]->buffer().dim[0].extent == mStateSize);
+      MNN_ASSERT(weightsVec[i]->buffer().dim[1].extent == mStateSize);
+      trimTensor(weightsVec[i].get(), mWeights[i].get());
+    }
+    for (int i = 8 + b * 12; i < 12 + b * 12; i++) {
+      // Bi, Bn, Bf, Bo
+      MNN_ASSERT(weightsVec[i]->dimensions() == 1);
+      MNN_ASSERT(weightsVec[i]->buffer().dim[0].extent == mStateSize);
+      trimTensor(weightsVec[i].get(), mWeights[i].get());
+    }
+  }
+  return NO_ERROR;
+}
+
+ErrorCode BlstmComputer::onResize(int timeSteps, int batchSize) {
+  if (batchSize != mBatchSize) {
+    // Reinitialize mHiddenStates & mCellStates
+    mHiddenStates.clear();
+    mCellStates.clear();
+    for (int i = 0; i < (mBidirectional ? 2 : 1); i++) {
+      mHiddenStates.push_back(shared_ptr<Tensor>(Tensor::createDevice<float>(
+          vector<int>{batchSize, mStateSize}, Tensor::CAFFE)));
+      backend()->onAcquireBuffer(mHiddenStates[i].get(), Backend::DYNAMIC);
+      mCellStates.push_back(shared_ptr<Tensor>(Tensor::createDevice<float>(
+          vector<int>{batchSize, mStateSize}, Tensor::CAFFE)));
+      backend()->onAcquireBuffer(mCellStates[i].get(), Backend::DYNAMIC);
+    }
+  }
+
+  if (batchSize != mBatchSize || timeSteps != mTimeSteps) {
+    // Reinitialize mInput, mGateInputs, mGateOutputs, mOutput
+    mInput.reset(Tensor::createDevice<float>(
+        vector<int>{batchSize, timeSteps, mInDim}, Tensor::CAFFE));
+    backend()->onAcquireBuffer(mInput.get(), Backend::DYNAMIC);
+
+    mGateInputs.clear();
+    mGateOutputs.clear();
+    for (int i = 0; i < (mBidirectional ? 8 : 4); i++) {
+      mGateInputs.push_back(shared_ptr<Tensor>(Tensor::createDevice<float>(
+          vector<int>{batchSize * timeSteps, mStateSize}, Tensor::CAFFE)));
+      backend()->onAcquireBuffer(mGateInputs[i].get(), Backend::DYNAMIC);
+      mGateOutputs.push_back(shared_ptr<Tensor>(Tensor::createDevice<float>(
+          vector<int>{batchSize, mStateSize}, Tensor::CAFFE)));
+      backend()->onAcquireBuffer(mGateOutputs[i].get(), Backend::DYNAMIC);
+    }
+
+    mOutput.reset(Tensor::createDevice<float>(
+        vector<int>{batchSize * timeSteps,
+                    mBidirectional ? 2 * mStateSize : mStateSize},
+        Tensor::CAFFE));
+    backend()->onAcquireBuffer(mOutput.get(), Backend::DYNAMIC);
+  }
+  mBatchSize = batchSize;
```

Comment:
After the mid tensor usage, it should be released by onReleaseBuffer

Reply:
Thanks, my fault. Already resolved

Reply:
Thanks.